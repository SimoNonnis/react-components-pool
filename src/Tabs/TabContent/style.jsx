import styled from 'styled-components';

export const Container = styled.div`
  min-height: 200px;
  background-color: #fff;
  padding: 1.5em;
`;
