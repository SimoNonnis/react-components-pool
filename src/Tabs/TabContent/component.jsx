import PropTypes from 'prop-types';
import React from 'react';

import { Container } from './style';

/**
 * TabContent renders the content of the tab, wrapped in a div.
 * @param {{children: React.Component}} children
 * @return {React.Component}
 */
const TabContent = ({ children }) => <Container>{children}</Container>;

TabContent.propTypes = {
  children: PropTypes.node.isRequired,
};

export default TabContent;
