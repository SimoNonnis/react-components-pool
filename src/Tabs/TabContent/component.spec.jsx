/* eslint-env jest */
import React from 'react';

import 'jest-styled-components';

import TabContent from './component';

describe('Test <TabContent />', () => {
  const wrapper = shallow(<TabContent>The 1st tab content is here</TabContent>);

  it('should find <TabContent /> component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
