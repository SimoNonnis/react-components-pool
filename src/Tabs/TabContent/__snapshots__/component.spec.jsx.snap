// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <TabContent /> should find <TabContent /> component 1`] = `
<styled.div>
  The 1st tab content is here
</styled.div>
`;
